plugin/adsk/scripts/USDMenuProc.mel

```diff
@@ -55,6 +55,117 @@ global proc mayaUsdMenu_duplicate(string $ufePath)
     }
 }
 
+global proc mayaUsdMenu_assignNewMaterial(string $ufePath, string $material)
+{
+    if (!hasPrimUpdater())
+        return;
+
+    if (size($ufePath) != 0 && size($material) != 0) {
+        string $temp = `python("import ufe;\
+                                ufePath = ufe.PathString.path('" + $ufePath + "');\
+                                item = ufe.Hierarchy.createItem(ufePath);\
+                                contextOps = ufe.ContextOps.contextOps(item);\
+                                cmd = contextOps.doOpCmd(['Assign New Material', '', '" + $material + "']);\
+                                cmd.execute();")`;
```

Comment:
This is where you need to use `Ufe::UndoableCommandMgr::instance().executeCmd(cmd);`